stuff/cron/update_ranks.py

```diff
@@ -174,6 +174,60 @@ def update_user_rank(cur: MySQLdb.cursors.BaseCursor) -> Sequence[float]:
     return scores
 
 
+def update_author_rank(cur: MySQLdb.cursors.BaseCursor) -> None:
+    '''Updates the author's ranking'''
+    logging.info('Updating authors ranking...')
+    cur.execute('''
+        SELECT
+            `u`.`user_id`,
+            `i`.`username`,
+            `i`.`name`,
+            `i`.`country_id`,
+            `i`.`state_id`,
+            `isc`.`school_id`,
+            SUM(`p`.`quality`) AS `author_score`
+        FROM
+            `Problems` AS `p`
+        INNER JOIN
+            `ACLs` AS `a` ON `a`.`acl_id` = `p`.`acl_id`
+        INNER JOIN
+            `Users` AS `u` ON `u`.`user_id` = `a`.`owner_id`
+        INNER JOIN
+            `Identities` AS `i` ON `i`.`identity_id` = `u`.`main_identity_id`
+        LEFT JOIN
+            `Identities_Schools` AS `isc`
+        ON
+            `isc`.`identity_school_id` = `i`.`current_identity_school_id`
+        WHERE
+            `p`.`quality` IS NOT NULL
+        GROUP BY
+            `u`.`user_id`
+        ORDER BY
+            `author_score`
+    ''')
+
+    prev_score = None
+    rank = 0
+    for index, row in enumerate(cur):
+        logging.info(row)
+        if row['author_score'] != prev_score:
+            rank = index + 1
+        prev_score = row['author_score']
+        cur.execute('''
+                    INSERT INTO
+                        `User_Rank` (`user_id`, `username`, `author_score`,
+                                     `author_ranking`, `name`, `country_id`,
+                                     `state_id`, `school_id`)
+                    VALUES (%s, %s, %s, %s, %s, %s, %s, %s)
+                    ON DUPLICATE KEY
+                        UPDATE
+                            author_ranking = %s,
+                            author_score = %s;''',
```

Comment:
aquí funciona [`REPLACE INTO`](https://dev.mysql.com/doc/refman/8.0/en/replace.html)

```suggestion
                    REPLACE INTO
                        `User_Rank` (`user_id`, `username`, `author_score`,
                                     `author_ranking`, `name`, `country_id`,
                                     `state_id`, `school_id`)
                    VALUES (%s, %s, %s, %s, %s, %s, %s, %s);'',
```

?

Reply:
No funciona:
```
REPLACE works exactly like INSERT, except that if an old row in the table has the same value as a new row for a PRIMARY KEY or a UNIQUE index, the old row is deleted before the new row is inserted. See Section 13.2.6, “INSERT Statement”.
```

Borra el registro anterior, y por lo tanto, se pierde el rank y score de usuario.

Reply:
va, gracias por confirmar!